tests/test_payment_overdue.py

```
@@ -0,0 +1,110 @@
+
+
+def test_overdue_payment_cancels_relationship(test_client, create_client, coach_auth_header, db_session):
+    """
+    When refresh_payments runs and the most recent billing cycle has already
+    expired (end_date < today) with an unpaid balance, it should:
+    - Cancel the subscription
+    - Terminate the coach-client relationship
+    - Notify both the client and coach
+    - NOT create a new billing cycle
+    """
+    os.environ["CRON_SECRET"] = "test-cron-secret"
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to: "CRON_SECRET"

(S105)